proto/buf/alpha/registry/v1alpha1/labels.proto

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+enum LabelNamespace {
+  LABEL_NAMESPACE_UNSPECIFIED = 0;
+  LABEL_NAMESPACE_TAG = 1;
+  LABEL_NAMESPACE_BRANCH = 2;
+  LABEL_NAMESPACE_GIT_COMMIT = 3;
+}
+
+message Label {
+  LabelNamespace namespace = 1;
+  string name = 2;
+  LabelValue label_value = 3;
+}
+
+message LabelValue {
+  string commit_id = 1;
+}
+
+service LabelService {
+  rpc CreateLabel(CreateLabelRequest) returns (CreateLabelResponse);
+  rpc MoveLabel(MoveLabelRequest) returns (MoveLabelResponse);
+  rpc GetLabels(GetLabelsRequest) returns (GetLabelsResponse);
+}
+
+message CreateLabelRequest {
+  Label label = 1;
+  optional string author = 2;
+  optional google.protobuf.Timestamp create_time = 3;
+}
+
+message CreateLabelResponse {
+  string commit_id = 1;
+}
+
+message MoveLabelRequest {
+  Label label = 1;
+  LabelValue from = 2;
+  LabelValue to = 3;
+}
+
+message MoveLabelResponse {}
+
+message GetLabelsRequest {
+  string name = 1;
+  string module_entity_id = 2;
```

Comment:
`buf.alpha.registry.v1alpha1.Repository` seems more appropriate?

Reply:
Also, a repo reference needs to be present in all the calls.

Reply:
I guess that's true. I'll have to check what our indexing looks like there. Conceptually I was always scoping commit identifiers to a repo, but these ids are supposed to be globally unique.

Related, this module_entity_id is yet another way of scoping to a repository. The type I suggested is pretty big and other APIs represent it differently. Let's chat about which pattern we should go with and avoid introducing a third (or even fourth) way to represent a repo address on the wire.

Reply:
Yeah -- we should definitely chat about this. I was a little bit concerned about bringing in the large types (I also thought about the `Reference` type in https://github.com/bufbuild/buf/blob/dce75c97b1ed32ffe321916ad611d32e0cad6737/proto/buf/alpha/registry/v1alpha1/reference.proto#L22-L36 but I had a couple of concerns regarding the scope of this type, so I ended up not doing that).